datafusion/datasource/src/file_scan_config.rs

```diff
@@ -326,14 +544,15 @@ impl FileScanConfig {
     /// # Parameters:
     /// * `object_store_url`: See [`Self::object_store_url`]
     /// * `file_schema`: See [`Self::file_schema`]
+    #[allow(deprecated)] // `new` will be removed same time as `with_source`
```

Comment:
this is nice that it will deprecate the old API rather and will guide users during upgrade